src/main/java/zmq/Ctx.java

```diff
@@ -149,6 +149,8 @@ private void destroy() throws IOException
         termMailbox.close();
 
         tag = 0xdeadbeef;
+
+        zmq.ZMQ.closePollSelector();
```

Comment:
I'm also late...
I would tend to give it an historical reason, @miniway could tell us more about that.
The tests I made on this library advocate against the use of ThreadLocal and finalize() to release the selector resource. This was also one big motivation to create the ZProxy class: having a better control on the proxy operations and suppress this memory leak for my own uses.